File: 0144-binary-tree-preorder-traversal/0144-binary-tree-preorder-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> preorderTraversal(TreeNode root) {
        return helper(root, new ArrayList<Integer>());
    }

    public List<Integer> helper(TreeNode cur, List<Integer> x){
        if (cur == null) return x;
        x.add(cur.val);
        helper(cur.left, x);
        helper(cur.right, x);
        return x;
    }
}

File: 0144-binary-tree-preorder-traversal/0144-binary-tree-preorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def preorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        a = []
        def dfsp(list, root) :
            if root is None :
                return
            list.append(root.val)
            dfsp(list, root.left)
            dfsp(list, root.right)
        dfsp(a, root)
        return a